netty-reactive-streams/src/main/java/com/typesafe/netty/HandlerPublisher.java

```diff
@@ -203,32 +203,40 @@ private void provideSubscriber(Subscriber<? super T> subscriber) {
 
     @Override
     public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-        verifyRegisteredWithRightExecutor(ctx);
+        // If the channel is not yet registered, then it's not safe to invoke any methods on it, eg read() or close()
+        // So don't provide the context until it is registered.
+        if (ctx.channel().isRegistered()) {
+            provideChannelContext(ctx);
+        }
```

Comment:
I see,thanks.but why not we only register the channel to the eventloop after there is an subscriber?and then we could fix this up.
I mean we fire the `subscriber.onSubscribe()` when we registered to the eventloop,eg,inside the ChannelFuture's onComplete callback to invoke it.